qbraid/transpiler/conversions/openqasm3/openqasm3_to_cudaq.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (C) 2024 qBraid
+#
+# This file is part of the qBraid-SDK
+#
+# The qBraid-SDK is free software released under the GNU General Public License v3
+# or later. You can redistribute and/or modify it under the terms of the GPL v3.
+# See the LICENSE file in the project root or <https://www.gnu.org/licenses/gpl-3.0.html>.
+#
+# THERE IS NO WARRANTY for the qBraid-SDK, as per Section 15 of the GPL v3.
+
+"""
+Module containing OpenQASM to CUDA-Q conversion function
+
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Dict, Union
+
+import pyqasm
+from openqasm3 import ast, parser, printer
+from qbraid_core._import import LazyLoader
+
+from qbraid.passes.qasm import normalize_qasm_gate_params
+from qbraid.transpiler.annotations import requires_extras, weight
+from qbraid.transpiler.exceptions import ProgramConversionError
+
+cudaq = LazyLoader("cudaq", globals(), "cudaq")
+
+if TYPE_CHECKING:
+    from cudaq import PyKernel, QuakeValue
+
+
+def gate_kernel(name: str, *args) -> PyKernel:
+    """Returns CUDA-Q kernel for pure standard gates (no modifiers - ctrl or adj)."""
+
+    if name in ["x", "y", "z", "h", "s", "t", "sdg", "tdg", "i", "id", "iden"]:
+        size, argc = 1, 0
+    elif name in ["rx", "ry", "rz"]:
+        size, argc = 1, 1
+    elif name in ["swap"]:
+        size, argc = 2, 0
+    elif name in ["u3"]:
+        size, argc = 3, 3
+    else:
+        raise ProgramConversionError(f"Unsupported gate: {name}")
+
+    kernel, *qrefs = cudaq.make_kernel(*[cudaq.qubit for _ in range(size)])
+
+    if len(args) != argc:
+        raise ProgramConversionError(
+            f"Gate {name} requires {argc} args but only {len(args)} were provided: {args}"
+        )
+
+    if name in ["i", "id", "iden"]:
+        return kernel
+
+    op = getattr(kernel, name)
+
+    if len(args) > 0:
+        op(*args, *qrefs)
+    else:
+        op(*qrefs)
+
+    return kernel
+
+
+def identifier_lookup(cbit: Union[ast.IndexedIdentifier, ast.Identifier]) -> str:
+    """Unpacks the identifier name"""
+    if isinstance(cbit, ast.IndexedIdentifier):
+        return cbit.name.name
+
+    return cbit.name
+
+
+@weight(1)
+@requires_extras("cudaq")
+# pylint: disable-next=too-many-statements
+def openqasm3_to_cudaq(program: ast.Program) -> PyKernel:
+    """Returns a CUDA-Q kernel representing the input OpenQASM program.
+
+    Args:
+        qasm (str or ast.Program): OpenQASM program to convert to CUDA-Q kernel.
+
+    Returns:
+        kernel: CUDA-Q kernel equivalent to input OpenQASM string.
+    """
+    try:
+        module = pyqasm.load(program)
+        module.validate()
+    except Exception as e:
+        raise ProgramConversionError("PyQasm program is not well-formed") from e
+
+    module.unroll()
+    program = module.unrolled_ast
+    program = parser.parse(normalize_qasm_gate_params(printer.dumps(program)))
```

Comment:
You don't need to add a call to `normalize_qasm_gate_params` as `unroll` operation by default substitutes the values of all variables inside the unrolled qasm.